.github/workflows/build_and_test.yml

```diff
@@ -556,8 +581,14 @@ jobs:
           submodules: true
           fetch-depth: 0
 
+      - name: Setup python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
```

Comment:
Can we use `rust` image here which comes with python?

Reply:
To be done after this is merged as that would be a larger refactoring of the deployment script